tests/test_execution/test_backends.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from unittest.mock import MagicMock
+
+import pytest
+from pulser.backend import BitStrings, EmulationConfig, EmulatorBackend, Results
+from pulser.backend.remote import JobParams, RemoteConnection, RemoteResults
+from pulser.sequence import Sequence as PulserSequence
+from pulser_pasqal.backends import RemoteEmulatorBackend
+
+from qoolqit.execution import LocalEmulator, RemoteEmulator
+from qoolqit.program import QuantumProgram
+
+
+class TestBackends:
```

Comment:
I believe that it would be useful to test using Pasqal Cloud's MockServer.
https://github.com/pasqal-io/pasqal-cloud/blob/main/pasqal-cloud/pasqal_cloud/utils/mock_server.py .

See in `test_qoolqit_backends.py` `MyMockServer` (you don't need `BaseMockServer` anymore, it's now part of `pasqal-cloud`) for some properties we're testing using it (in particular that `wait` is used correctly).

@MatthieuMoreau0, do you have other suggestions for tests we could add here?